Sources/SwiftRepo/Repository/StoreModel.swift

```diff
@@ -0,0 +1,28 @@
+//
+//  StoreModel.swift
+//  SwiftRepo
+//
+//  Created by Carter Foughty on 10/1/24.
+//
+
+import Foundation
+
+/// A model that can be placed and fetched in a store.
+public protocol StoreModel {
+    /// The type to use as the identifier for the model
+    associatedtype Key = any Hashable
+
+    /// The identifier of the model
+    var id: Key { get }
+    /// A predicate that can be used to query for the `StoreModel`
+    static func predicate(key: Key) -> Predicate<Self>
+}
+
+public extension StoreModel where Key == UUID {
```

Comment:
Can this be abstracted to something like `Key: Equatable`? If not, how about if we add the usual suspect here, such as `String`, etc.

Reply:
Unfortunately, predicates aren't happy with generics, so I added extensions for obvious types.

Reply:
Figures